syntax = "proto3";
package feast.core;

option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
option java_outer_classname = "FeatureReferenceProto";
option java_package = "feast.proto.core";

import "feast/core/Feature.proto";
import "feast/core/DataSource.proto";


// A projection to be applied on top of a FeatureView. 
// Contains the modifications to a FeatureView such as the features subset to use.
message FeatureViewProjection {
  // The feature view name
  string feature_view_name = 1;

  // Alias for feature view name
  string feature_view_name_alias = 3;

  // The features of the feature view that are a part of the feature reference.
  repeated FeatureSpecV2 feature_columns = 2;

  // Map for entity join_key overrides of feature data entity join_key to entity data join_key
  map<string,string> join_key_map = 4;

  string timestamp_field = 5;
  string date_partition_column = 6;
  string created_timestamp_column = 7;
  // Batch/Offline DataSource where this view can retrieve offline feature data.
  DataSource batch_source = 8;
  // Streaming DataSource from where this view can consume "online" feature data.
  DataSource stream_source = 9;

}
